pystiche_papers/sanakoyeu_et_al_2018/_loss.py

```diff
@@ -289,3 +293,94 @@ def transformed_image_loss(
     return ops.FeatureReconstructionOperator(
         transformer_block, score_weight=score_weight
     )
+
+
+class FeatureReconstructionOperator(ops.FeatureReconstructionOperator):
+    def __init__(
+        self,
+        encoder: SequentialEncoder,
+        score_weight: float = 1.0,
+        impl_params: bool = True,
+    ):
+        super().__init__(encoder, score_weight=score_weight)
+        self.impl_params = impl_params
+
+    def calculate_score(  # type: ignore[override]
+        self, input_repr: torch.Tensor, target_repr: torch.Tensor, ctx: None
+    ) -> torch.Tensor:
+        if self.impl_params:
+            # https://github.com/pmeier/adaptive-style-transfer/blob/07a3b3fcb2eeed2bf9a22a9de59c0aea7de44181/model.py#L194
+            # https://github.com/pmeier/adaptive-style-transfer/blob/07a3b3fcb2eeed2bf9a22a9de59c0aea7de44181/module.py#L177-L178
+            return torch.mean(torch.abs(input_repr - target_repr))
```

Comment:
Changing the reduction is not a change that should happen here since it is a core concept of the `ops.FeatureReconstructionOperator`. I suggest you implement an `MAEReconstructionOperator` and return this or the other based on `impl_params` from `style_aware_content_loss`.